tests/integration/VideoInputTest.ts

```diff
@@ -0,0 +1,65 @@
+/* tslint:disable only-arrow-functions */
+
+import * as assert from 'assert';
+import * as sinon from 'sinon';
+
+import {
+  testVideoInputDevice,
+  VideoInputTest,
+} from '../../lib/VideoInputTest';
+
+const defaultTestDuration = 500;
+
+describe('testVideoInputDevice', function() {
+  let errorHandler: sinon.SinonSpy;
+  let endHandler: sinon.SinonSpy;
+
+  beforeEach(async function() {
+    errorHandler = sinon.spy();
+    endHandler = sinon.spy();
+
+    await new Promise(resolve => {
+      const test = testVideoInputDevice({
+        duration: defaultTestDuration,
+      });
+      test.on(VideoInputTest.Events.Error, errorHandler);
+      test.on(VideoInputTest.Events.End, (r) => {
+        endHandler(r);
+        // we want to wait before resolving so we can detect if the end handler
+        // has been called multiple times
+        setTimeout(() => resolve(), defaultTestDuration);
+      });
+    });
+  });
+
+  it('should have not called the error handler', function() {
+    assert(!errorHandler.called);
+  });
+
+  it('should have called the end handler once', function() {
+    assert(endHandler.callCount === 1);
```

Comment:
We get more descriptive error messages when we use `assert.equal(endHandler.callCount, 1)` in cases like these :)